unit_tests/snapshot_interpolator_test.cpp

```diff
@@ -0,0 +1,122 @@
+                       -0.288488198334352, -0.697095554949408, -0.939878053603591, -0.782971083698006, -0.279415498198926,
+                       0.188293444380395, 0.669217685146469, 0.965688937709033, 0.846474732609843, 0.412118485241757,
+                       -0.077580693288345, -0.623537711025344, -0.971758328554163, -0.890773577229575, -0.536572918000435,
+                       -0.041614069121016, 0.560852411851254, 0.957953731078007, 0.938810668593539, 0.650287840157117};
+
+    std::vector<Vector*> snapshots;
+    std::vector<Vector*> out_snapshots;
+    std::vector<Vector*> reference_snapshots;
+    std::vector<Vector*> times;
+    std::vector<Vector*> out_times;
+    for(int i = 0; i < t.size(); ++i)
+    {
+        Vector* temp_t = new Vector(1, false);
+        temp_t->item(0) = t[i];
+        times.push_back(temp_t);
+        Vector* temp_y = new Vector(2,false);
+        temp_y->item(0) = y[i];
+        temp_y->item(1) = y2[i];
+        snapshots.push_back(temp_y);
+    }
+
+    for(int i = 0; i < tq.size(); ++i)
+    {
+        Vector* temp_t = new Vector(1, false);
+        temp_t->item(0) = tq[i];
+        out_times.push_back(temp_t);
+        Vector* temp_y = new Vector(2,false);
+        temp_y->item(0) = yq1[i];
+        temp_y->item(1) = yq2[i];
+        reference_snapshots.push_back(temp_y);
+    }
+
+    SnapshotInterpolator* interp = new SnapshotInterpolator();
+
+    std::cout << "Beginning base interpolation function" << std::endl;
+    out_snapshots = interp->interpolate(times,snapshots,out_times);
+    std::cout << "Finished interpolation " << std::endl;
+    /*
+       for(int i = 0; i < out_snapshots.size(); ++i)
+        {
+            std::cout << "Time " << i << " is " << out_times[i]->item(0) << std::endl;
+            std::cout << "Reference at " << i << " is (" << reference_snapshots[i]->item(0) <<
+                         "," << reference_snapshots[i]->item(1) << ")" << std::endl;
+            std::cout << "Snapshot interpolation at " << i << " is (" << out_snapshots[i]->item(0) <<
+                         "," << out_snapshots[i]->item(1) << ")" << std::endl;
+        }
+     */
+    for(int i = 0; i < out_snapshots.size(); ++i)
+    {
+        std::cout << "Error at time[" << i << "] = " << out_times[i]->item(
+                      0) << " is (" <<
+                  reference_snapshots[i]->item(0) - out_snapshots[i]->item(0) << ","
+                  << reference_snapshots[i]->item(1) - out_snapshots[i]->item(
+                      1) << ") " <<  std::endl;
+    }
+
+
+    std::cout << "Beginning variant interpolation function" << std::endl;
+    out_snapshots = interp->interpolate(times,snapshots,26,&out_times);
+    std::cout << "Finished interpolation " << std::endl;
+
+    for(int i = 0; i < out_snapshots.size(); ++i)
+    {
+        std::cout << "Error at time[" << i << "] = " << out_times[i]->item(
```

Comment:
Instead of writing output, a unit test should check that the results are correct. For example, see `EXPECT_DOUBLE_EQ` in `test_Matrix.cpp`.

Reply:
I have updated this test to print only a single line whether the "Test Succeeded" or the "Test Failed".  If there is some other convention here, I am happy to update.